File: skills/invoice-conversion/examples/zugferd-to-json.py
#!/usr/bin/env python3
# Extract Invoice JSON from a ZUGFeRD PDF
# Usage: python zugferd-to-json.py <invoice.zugferd.pdf>
#
# Requires: INVAPI_API_KEY environment variable, requests library

import os
import sys

import requests

if len(sys.argv) < 2:
    print("Usage: python zugferd-to-json.py <invoice.zugferd.pdf>", file=sys.stderr)
    sys.exit(1)

input_path = sys.argv[1]

api_key = os.environ.get("INVAPI_API_KEY")
if not api_key:
    print("Error: INVAPI_API_KEY environment variable is not set.", file=sys.stderr)
    sys.exit(1)

with open(input_path, "rb") as f:
    body = f.read()

resp = requests.post(
    "https://api.invapi.org/api/v1/zugferd/json",
    headers={"x-api-key": api_key, "Content-Type": "application/pdf"},
    data=body,
)
resp.raise_for_status()
print(resp.text)

File: skills/invoice-conversion/examples/zugferd-to-json.js
#!/usr/bin/env node
// Extract Invoice JSON from a ZUGFeRD PDF
// Usage: node zugferd-to-json.js <invoice.zugferd.pdf>
//
// Requires: INVAPI_API_KEY environment variable

const fs = require("fs");

const input = process.argv[2];
if (!input) {
  console.error("Usage: node zugferd-to-json.js <invoice.zugferd.pdf>");
  process.exit(1);
}

const apiKey = process.env.INVAPI_API_KEY;
if (!apiKey) {
  console.error("Error: INVAPI_API_KEY environment variable is not set.");
  process.exit(1);
}

const body = fs.readFileSync(input);

fetch("https://api.invapi.org/api/v1/zugferd/json", {
  method: "POST",
  headers: {
    "x-api-key": apiKey,
    "Content-Type": "application/pdf",
  },
  body,
})
  .then((res) => res.text())
  .then((text) => console.log(text))
  .catch((err) => {
    console.error(err);
    process.exit(1);
  });
